src/Momento.Sdk/Config/Transport/IGrpcConfiguration.cs

```diff
@@ -42,6 +43,36 @@ public interface IGrpcConfiguration
     /// </remarks>
     public SocketsHttpHandlerOptions SocketsHttpHandlerOptions { get; }
 
+    /// <summary>
+    /// Override the time to wait for a response from a keepalive or ping.
+    /// NOTE: keep-alives are very important for long-lived server environments where there may be periods of time
+    /// when the connection is idle. However, they are very problematic for lambda environments where the lambda
+    /// runtime is continuously frozen and unfrozen, because the lambda may be frozen before the "ACK" is received
+    /// from the server. This can cause the keep-alive to timeout even though the connection is completely healthy.
+    /// Therefore, keep-alives should be disabled in lambda and similar environments.
+    /// </summary>
+    public TimeSpan KeepAlivePingTimeout { get; }
+
+    /// <summary>
+    /// After a duration of this time the client/server pings its peer to see if the transport is still alive.
+    /// NOTE: keep-alives are very important for long-lived server environments where there may be periods of time
+    /// when the connection is idle. However, they are very problematic for lambda environments where the lambda
+    /// runtime is continuously frozen and unfrozen, because the lambda may be frozen before the "ACK" is received
+    /// from the server. This can cause the keep-alive to timeout even though the connection is completely healthy.
+    /// Therefore, keep-alives should be disabled in lambda and similar environments.
+    /// </summary>
+    public TimeSpan KeepAlivePingDelay { get; }
+
+    /// <summary>
+    /// Indicates if it permissible to send keepalive pings from the client without any outstanding streams.
+    /// NOTE: keep-alives are very important for long-lived server environments where there may be periods of time
+    /// when the connection is idle. However, they are very problematic for lambda environments where the lambda
+    /// runtime is continuously frozen and unfrozen, because the lambda may be frozen before the "ACK" is received
+    /// from the server. This can cause the keep-alive to timeout even though the connection is completely healthy.
+    /// Therefore, keep-alives should be disabled in lambda and similar environments.
+    /// </summary>
+    public bool KeepAlivePermitWithoutCalls { get; }
```

Comment:
Using type `bool` here because I could not get the build to recognize the `System.Net.Http.HttpKeepAlivePingPolicy` enum that's actually used when setting the keepalive option in the `SocketsHttpHandler` in the `Utils.GrpcChannelOptionsFromGrpcConfig` function

Reply:
bool sounds okay to me, in general I usually try to avoid leaking types from dependency libraries into the public API of my own library. We have already done that with ChannelOptions so it's not a completely black and white thing, but I think a boolean is fine since we are modeling our own config fields here.

Reply:
re: build failures -- .NET has portability issues because different runtimes support different HttpHandlers. For this reason I also recommend putting handler-specific settings in a handler-specific options type.

To access the enum type then we would have to add the conditional compilation guard around it. See the conditional compilation guards around the `SocketHttpHandler` code for reference